runtime/src/system_instruction_processor.rs

```diff
@@ -112,14 +115,20 @@ fn finish_create_account(
         debug!("Assign: program id {} invalid", program_id);
         return Err(SystemError::InvalidProgramId.into());
     }
-    to.account.owner = *program_id;
 
-    from.account.lamports -= lamports;
-    to.account.lamports += lamports;
-    to.account.data = vec![0; data_length as usize];
-    to.account.executable = false;
+    match get_system_account_kind(&from.account) {
```

Comment:
ok.  this could be simpler, more future-proof, and tested earlier (like at the top of this fn, when we're looking at `from` for other stuff.  something like:?

```
if from.data.len() != 0 {
    return Err(InstructionError::InvalidArgument);
}
```

Reply:
Sure. I was aiming more for the strictness and self-documenting properties of the helper, but am not opposed to changing that if you feel strongly. I'll move it up and simplify in either case

Reply:
I think from.data.len() != 0 is stricter ;)

Reply:
Ok, ok 🙂

Reply:
Added in https://github.com/solana-labs/solana/pull/7645/commits/50534076799eec7898caebea0ed7bdb6bfedf56b?w=1